viewing_party/party.py

```diff
@@ -0,0 +1,140 @@
+def create_movie(movie_title, genre, rating):
+    new_movie = {
+        "title": movie_title,
+        "genre": genre,
+        "rating": rating
+    }
+
+    for key in new_movie:
+        if new_movie[key] == None:
+            return None
+    return new_movie
+
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+
+def watch_movie(user_data, movie_title):
+    for movie in user_data["watchlist"]:
+        if movie_title == movie["title"]:
+            user_data["watched"].append(movie)
+            user_data["watchlist"].remove(movie)
+    
+    return user_data
+
+
+def get_watched_avg_rating(user_data):
+    avg_rating = 0.0
+    total_rating = 0.0
+    
+    for movie in user_data["watched"]:
+        total_rating += movie["rating"] 
+    
+    if len(user_data["watched"]) > 0:
+        avg_rating = total_rating/len(user_data["watched"])
+    return avg_rating
+
+
+def get_most_watched_genre(user_data):
+    if not user_data["watched"]:
+        return None
```

Comment:
This is a nice use of what is called a guard clause or guard condition. It lets us check for a certain condition that has a specific result immediately before diving into the main part of our function. We can avoid using an else block (since the if condition, if taken, returns, turning the entire remainder of the function into one big else!).